### ISSUE TITLE
vips_image_new_temp_file() and vips_image_write() to flush image to disk causes issues

### ISSUE THREAD

**Issue by Warren-GH** [jcupitt/libvips#708]
For some reason I get errors and crashes if I do the following in an attempt to flush results of vips_mosaic/vips_merge to disk to prevent out of memory (stitching 300k x 300k image). I am using vips 8.5.6 on Windows with your precompiled binaries through the C interface:

#if SWAP_TO_DISK
VipsImage* previous = result->Image;
VipsImage* temp = vips_image_new_temp_file("%s.v");

vips_image_write(previous, temp);

// Crashes later on if this line is uncommented ??
g_object_unref(previous);

result->Image = temp;
#endif

Any idea why the VipsImage* from temp file is not the 'same' as the one that was generated by vips_mosaic/vips_merge? I have plenty of disk space, and memory.

More info:

Precursor errors look like:

Glib-GObject-WARNING **: gsignal.c:3183: signal id '6' is invalid for instance '0000027EBFAA4C80'
Glib-GObject-WARNING **: gsignal.c:3183: signal id '7' is invalid for instance '0000027EBFAA4C80'
[Above repeated many times]

Then crash. Still working on this but perhaps some outside perspective will help.

**Comment by jcupitt** [comment]
Hello Warren, yes, I think `vips_image_write()` is not handling this case well, I'll see what I can do.

In the meantime, I would just write to a file, eg.

```C
VipsImage *huge = ....;

if( vips_image_write_to_file( huge, "temp.v", NULL ) )
    error(...);
if( !(huge = vips_image_new_from_file( "temp.v", NULL )) )
    error(...);
```

You'll need to somehow remove `temp.v` when you're done.

**Comment by Warren-GH** [comment]
Thanks for the quick response. I changed it to the following in an attempt to mimic the same behavior as vips_image_new_temp_file(). It is missing the behavior of deleting the files if the program gets interrupted but otherwise I think it is working. 



    #if SWAP_TO_DISK
		VipsImage* previous = result->Image;
		char* tempName = vips__temp_name("%s.v");

		vips_image_write_to_file(previous, tempName, 0);
		g_object_unref(previous);

		VipsImage* fileImage = vips_image_new_from_file(tempName, 0);

		g_free(tempName);
		assert(fileImage);

		vips_image_set_delete_on_close(fileImage, true);

		result->Image = fileImage;
    #endif

**Comment by jcupitt** [comment]
I think I found a nice fix for this issue, it'll be in 8.6.

Thanks for reporting this, Warren.
